### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Denisprkh**

### CODE REVIEW

**Comment by NatalliaKhadunai** [comment]
I'm not able to run gradle tasks from the project root.

P.s. I googled possible reasons for the issues and it looks like the reason is missing file `/epam-lab-esm/gradle/wrapper/gradle-wrapper.jar`.

**Note: please always double-check that your code can be built and deployed before submitting a PR!**

**Comment by NatalliaKhadunai** [comment]
Please provide sql file for database initialization.

**Comment by Denisprkh** [comment]
> I'm not able to run gradlew tasks from the project root. Please fix this.
> 
> P.s. I googled possible reasons for the issues and it looks like the reason is missing file `/epam-lab-esm/gradle/wrapper/gradle-wrapper.jar`.
> 
> **Note: please always double-check that your code can be built and deployed before submitting a PR!**

graddle-wrapper.jar is added

**Comment by Denisprkh** [comment]
> Please provide sql file for database initialization.

db creation script is added

**Comment by NatalliaKhadunai** [comment]
Please pay attention to formatting. At the moment there are a lot of missing spaces before curly braces, comas, etc.

**Review by NatalliaKhadunai — CHANGES_REQUESTED** [review]

File: dbinfo/esm.sql [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  `tag_id` INT NOT NULL,
+  `gift_certificate_id` INT NOT NULL,
+  PRIMARY KEY (`tag_id`, `gift_certificate_id`),
+  INDEX `fk_lnk_tags_gift_certificates_tags_idx` (`tag_id` ASC) VISIBLE,
+  INDEX `fk_lnk_tags_gift_certificates_gift_certificates1_idx` (`gift_certificate_id` ASC) VISIBLE,
+  CONSTRAINT `fk_lnk_tags_gift_certificates_tags`
+    FOREIGN KEY (`tag_id`)
+    REFERENCES `esm`.`tag` (`tag_id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
```
Comment (NatalliaKhadunai) [inline dbinfo/esm.sql:59]: `NO ACTION` is a default option for `ON DELETE` and `ON UPDATE` so we can remove these lines everywhere.

File: src/main/resources/database.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+user=root
+password=b9227a3182cfe9a4dee384ee47b803f7
+driver=com.mysql.cj.jdbc.Driver
+url=jdbc:mysql://localhost:3306/esm?serverTimezone=Europe/Moscow&characterEncoding=\
+  UTF-8&useSSL=TRUE&allowPublicKeyRetrieval=TRUE
```
Comment (NatalliaKhadunai) [inline src/main/resources/database.properties:5]: I think we can remove all the properties from url. I read the documentation and didn't find any proof that we need them and the application works fine without them.

File: src/main/java/com/epam/esm/repository/tag/TagSqlQuery.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.epam.esm.repository.tag;
+
+public final class TagSqlQuery {
+
+    public static final String ADD_TAG = "INSERT INTO tag (name) VALUES (?)";
+    public static final String FIND_TAG_BY_ID = "SELECT tag_id, name FROM tag WHERE tag_id = ?";
```
Comment (NatalliaKhadunai) [inline src/main/java/com/epam/esm/repository/tag/TagSqlQuery.java:6]: Maybe `SELECT *` ?

File: src/main/java/com/epam/esm/config/AppConfig.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    private String password;
+
+    public AppConfig(Environment environment) {
+        this.environment = environment;
+    }
+
+    @Bean
+    public DataSource dataSource() {
+        SimpleDriverDataSource dataSource = new SimpleDriverDataSource();
+        dataSource.setDriverClass(com.mysql.cj.jdbc.Driver.class);
```
Comment (NatalliaKhadunai) [inline src/main/java/com/epam/esm/config/AppConfig.java:40]: This is still implemented slightly incorrect. Here is how we should use `@Value`:
image

File: src/main/java/com/epam/esm/config/SpringMvcDispatcherServletInitializer.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.epam.esm.config;
+
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+
+public class SpringMvcDispatcherServletInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
```
Comment (NatalliaKhadunai) [inline src/main/java/com/epam/esm/config/SpringMvcDispatcherServletInitializer.java:5]: It looks like we can remove the class. App runs fine without it.

**Comment by Denisprkh** [comment]
Code is ready for the next review, @NatalliaKhadunai

**Pull request merged**